src/openloco/ui/scrollview.cpp

```diff
@@ -2,11 +2,161 @@
 #include "../interop/interop.hpp"
 #include "../ui.h"
 #include "WindowManager.h"
+#include <cmath>
 
 using namespace openloco::interop;
 
 namespace openloco::ui::scrollview
 {
+    static constexpr uint16_t thumbSize = 10;
+    static constexpr uint16_t barWidth = 11;
+
+    // 0x004C87E1
+    void scroll_5_follow(WindowType _pressedWindowType, window_number _pressedWindowNumber, widget_index _pressedWidgetIndex, size_t _currentScrollOffset, int16_t ax)
+    {
+        auto window = WindowManager::find(_pressedWindowType, _pressedWindowNumber);
+        if (window == nullptr)
+            return;
+
+        scroll_area_t& scrollArea = window->scroll_areas[_currentScrollOffset];
+        auto& w = window->widgets[_pressedWidgetIndex];
+
+        scrollArea.flags |= scroll_flags::HSCROLLBAR_THUMB_PRESSED;
+
+        uint16_t trackWidth = w.width() - 2 - thumbSize - thumbSize;
+        if (scrollArea.flags & scroll_flags::VSCROLLBAR_VISIBLE)
+        {
+            trackWidth -= barWidth;
+        }
+
+        float scrollStepSize = (float)scrollArea.contentWidth / (float)trackWidth;
+        ax = std::round(ax * scrollStepSize);
```

Comment:
I don't see the advantage over doing an integer division. Looking at https://gcc.godbolt.org/z/NoQOse, that generates an `idiv`, just like the assembly,

Reply:
More obvious/clear code when reading it. If it gets optimised away: all the better.

Reply:
These floats are not likely to be optimised away — that's why I'm commenting here. https://gcc.godbolt.org/z/ofG5kE